internal/ethereum/exec_query.go

```diff
@@ -117,37 +124,36 @@ func (c *ethConnector) callTransaction(ctx context.Context, tx *ethsigner.Transa
 
 	// some Ethereum implementations return revert reason data in the response's result object,
 	// check the output to see if there are error data and return proper errors
-	revertReason, ok := processRevertReason(ctx, outputData, errors)
+	revertReason := processRevertReason(ctx, outputData, errors)
 	if revertReason != "" {
-		if ok {
-			return nil, ffcapi.ErrorReasonTransactionReverted, i18n.NewError(ctx, msgs.MsgReverted, revertReason)
-		}
 		return nil, ffcapi.ErrorReasonTransactionReverted, i18n.NewError(ctx, msgs.MsgReverted, revertReason)
 	}
 
-	if method != nil {
-		// Parse the data against the outputs
-		var jsonData []byte
-		outputValueTree, err := method.Outputs.DecodeABIDataCtx(ctx, outputData, 0)
-		if err == nil {
-			// Serialize down to JSON, and wrap in a JSONAny
-			jsonData, err = c.serializer.SerializeJSONCtx(ctx, outputValueTree)
-		}
-		if err != nil {
-			log.L(ctx).Warnf("Invalid return data: %s", outputData)
-			return nil, "", i18n.NewError(ctx, msgs.MsgReturnDataInvalid, err)
-		}
-		return fftypes.JSONAnyPtrBytes(jsonData), "", nil
+	if method == nil {
+		// We got data back, but are ignoring at as we don't have a method to parse against
+		log.L(ctx).Warnf("Data returned from call with no method signature: %s", outputData.String())
+		return fftypes.JSONAnyPtr(fftypes.NullString), "", nil
 	}
 
-	return nil, mapError(callRPCMethods, rpcErr.Error()), rpcErr.Error()
+	// Parse the data against the outputs
+	var jsonData []byte
+	outputValueTree, err := method.Outputs.DecodeABIDataCtx(ctx, outputData, 0)
+	if err == nil {
+		// Serialize down to JSON, and wrap in a JSONAny
+		jsonData, err = c.serializer.SerializeJSONCtx(ctx, outputValueTree)
+	}
+	if err != nil {
+		log.L(ctx).Warnf("Invalid return data: %s", outputData)
+		return nil, "", i18n.NewError(ctx, msgs.MsgReturnDataInvalid, err)
+	}
+	return fftypes.JSONAnyPtrBytes(jsonData), "", nil
 }
 
 // processRevertReason returns under 3 different circumstances:
-// 1. non-empty string, true: valid reason has been successfully parsed
-// 2. non-empty string, false: error detail was present but failed to parse, string was raw data
-// 3. empty string, true: outputData is NOT an error detail data
-func processRevertReason(ctx context.Context, outputData ethtypes.HexBytes0xPrefix, errorAbis []*abi.Entry) (string, bool) {
+// 1. non-empty string - parsed by us: valid reason has been successfully parsed
+// 2. non-empty string - assumed to already be parsed by node: error detail was present but failed to parse, string was raw data
+// 3. empty string: outputData is NOT an error detail data
+func processRevertReason(ctx context.Context, outputData ethtypes.HexBytes0xPrefix, errorAbis []*abi.Entry) string {
```

Comment:
There were no code paths where we actually used the bool return value to change logic